src/lib/comms/sol-coap.c

```diff
@@ -899,6 +906,8 @@ sol_coap_send_packet_with_reply(struct sol_coap_server *server, struct sol_coap_
     }
 
     sol_coap_header_get_id(pkt, &reply->id);
+    reply->reentrant.in_use = false;
+    reply->reentrant.is_stale = false;
```

Comment:
optional: As `reply` is allocated with `calloc()` you don't need this. But if you think that this increases clarity, keep it.